flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/abilities/source/SourceAbilityContext.java

```diff
@@ -45,12 +45,16 @@
  * </ul>
  */
 public class SourceAbilityContext implements FlinkContext {
+
     private final RowType sourceRowType;
     private final FlinkContext context;
+    private final FlinkTypeFactory typeFactory;
 
-    public SourceAbilityContext(FlinkContext context, RowType sourceRowType) {
+    public SourceAbilityContext(
```

Comment:
nit: `FlinkContext context, FlinkTypeFactory typeFactory, RowType sourceRowType` it is otherwise weird to have the row type between context parameters